feat(curriculum): create tooling and deployment quiz (#56568)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-tooling-and-deployment/66fd6f0370213be90fed82ec.md

---

A development tool that writes bundles of javascript code for you

#### --answer--

A development tool that combines many JavaScript code files into a single one

### --question--

#### --text--

What type of graph does a bundler generate?

#### --distractors--

An unweighted graph

---

A weighted graph

---

An undependency graph

#### --answer--

A dependency graph

### --question--

#### --text--

Which of these is not an example of perfomance optimization feature of a bundler?

#### --distractors--

Code splitting

---

Hot module replacement

---

Minification

#### --answer--

Code generation

### --question--

#### --text--

The process of generating a relationship map of all the served files in a project is called?

#### --distractors--

Mappification

---

Relationship resolution

---

Data resolution

#### --answer--

Dependency resolution

### --question--

#### --text--

Which file does a bundler use to start traversing your code and understand its dependencies?

#### --distractors--

A CSS file

---

An `index.html` file

---

Any JavaScript file

#### --answer--

An Entry File

### --question--

#### --text--

> GitHub Actions offers features that let you _______

Reviewer: Not sure about this one. Technically all of the answers, except maybe user access control, would fit.

#### --distractors--

compiles JavaScript files

---

host static websites

---

manage user access control

#### --answer--

control deployments

### --question--

#### --text--

What feature of GitHub Actions allows you to automatically deploy a branch for review before merging?

#### --distractors--

Static hosting

---

Code linting

---

Branch restrictions

#### --answer--

Preview deployments

### --question--

#### --text--

Which tool can be used with GitHub to automate preview deployments for a pull request?

#### --distractors--

Webpack

---

Parcel

---

Docker

#### --answer--

GitHub Actions

### --question--

#### --text--

What file is required in a GitHub repository to trigger a preview deployment using GitHub Actions?

#### --distractors--

A Docker file

---

A `netlify.toml` file

---

A `README.md` file

#### --answer--

A `.github/workflows` YAML file

### --question--

#### --text--

> These are examples of events to trigger deployment workflow except ______

Reviewer: ```suggestion
Which of these is not an event that can trigger a GitHub Action?
```

#### --distractors--

`pull_request`

---

`push`

---

`workflow_dispatch`

#### --answer--

`commit`

### --question--

#### --text--

Which of the following actions is not an example of activating a serverless function?

#### --distractors--

Placing an e-commerce order

---

Uploading a text file

---

Signing up for a newsletter

#### --answer--

Navigating to a website

### --question--

#### --text--

Which of the following is not a common type of DNS record?

#### --distractors--

`A`

---

`MX`

---

`CNAME`

#### --answer--

`HTML`

### --question--

#### --text--

> What is the function of the post method in a form?

Reviewer: ```suggestion
What is the function of the `POST` method in a form?
```

#### --distractors--

For updating data

---

For deleting data

---

For reading data

#### --answer--

For creating data

### --question--

#### --text--

What is the primary purpose of an identity provider?

#### --distractors--

To store user data

---

To process payments

---

To host a website

#### --answer--

To authenticate users

### --question--

#### --text--

> Why is it important to integrate an identity provider in mordern application?

Reviewer: ```suggestion
Why is it important to integrate an identity provider in a modern application?
```

### --question--

#### --text--

In addition to authentictication, what other feature do identity providers offer?

#### --distractors--

Database management

---

User interface design

---

Content delivery

#### --answer--

Authorization control

### --question--

#### --text--

Which of the following platforms provides a built-in continuous deployment feature?

#### --distractors--

AWS EC2

---

GitHub Pages

---

VPS

#### --answer--

Netlify

### --question--

#### --text--

Which deployment platform is optimized for front-end projects like static websites?

#### --distractors--

AWS Lambda

---

VPS

---

Heroku

#### --answer--

Vercel

### --question--

#### --text--

Which type of server hosting allows you to fully control the server, including the operating system?

#### --distractors--

PaaS (Platform as a Service)

---

Serverless functions

---

SaaS (Software as a Service)

#### --answer--

VPS (Virtual Private Server)

### --question--

#### --text--

Which deployment option would you choose for full control over your deployment environment?

#### --distractors--

Vercel

---

Netlify

---

GitHub Pages

#### --answer--

> VPS

Reviewer: ```suggestion
VPS

```

Restore the EOF new line.

Reviewer 2: resolving the conversation doesn't mean that the issue is fixed, please actually do the thing @SelormDev

Author: Pardon me but I thought commiting the suggestion will apply the changes requested. Let me manually check and do the necessary changes.

Reviewer 2: the diff still shows a removed line

![image](https://github.com/user-attachments/assets/608c00e4-2338-45fc-8624-ffe47e2a5cfc)

Reviewer 2: now it's fine, I committed the suggestion
